### PR TITLE
Compute WER, MER, WIL and WIP

### LINKED ISSUE #9: WIL and MER
Additionally to WER, WIL (Word Information Lost) and MER (Match Error Rate), can be good performance measures for ASR. I could not find any pypi packages that support those measures, and think they might be a good fit for **jiwer**.

See:
- [Automatic Speech Recognition
Errors Detection and Correction: A Review](https://www.researchgate.net/profile/Rahhal_Errattahi/publication/324427003_Automatic_Speech_Recognition_Errors_Detection_and_Correction_A_Review/links/5ad462590f7e9b2859365080/Automatic-Speech-Recognition-Errors-Detection-and-Correction-A-Review.pdf)
- [From WER and RIL to MER and WIL: improved evaluation
measures for connected speech recognition](https://pdfs.semanticscholar.org/8516/531ff3bd874b66b811f0bd4e21a2d6b10e54.pdf)

I'd be happy to open a pull request with those measures added, I you think they could fit this package.

### PR DESCRIPTION
**Pull request by bartvanandel**
Implements a base method, `compute_measures`, for computing all mentioned measures, and convenience methods to extract the individual measures. Updated unit tests and docs as well. Fixes #9 

BTW, adapting `compute_measures` to provide information about hit, substitution, deletion and insertion counts is now trivial, in case this proves to be interesting.

### CODE REVIEW

File: jiwer/measures.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+################################################################################
+# Implementation of helper methods
+
+def _preprocess(
+    truth: Union[str, List[str]],
+    hypothesis: Union[str, List[str]],
+    truth_transform: Union[tr.Compose, tr.AbstractTransform],
+    hypothesis_transform: Union[tr.Compose, tr.AbstractTransform]
+) -> List[str]:
```
Comment (nikvaessen) [inline jiwer/measures.py:200]: Tuple[str, str]

File: README.md [hunk 2]
```diff
@@ -1,8 +1,11 @@
-# Word Error Rate for automatic speech recognition
+# Simularity error measures for automatic speech recognition
 
-This repository contains a simple python package to approximate the WER of a transcript. It computes the minimum-edit distance 
-between the ground-truth sentence and the hypothesis sentence of a speech-to-text API. The minimum-edit distance is calculated
-using the python C module [python-Levenshtein](https://github.com/ztane/python-Levenshtein).
+This repository contains a simple python package to approximate the Word Error Rate (WER), Match Error Rate (MER), Word Information Preserved (WIP) and Word Information Lost (WIP) of a transcript.
```
Comment (nikvaessen) [inline README.md:3]: "WIL" instead of "WIP a second time

File: jiwer/measures.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+This file implements methods for calculating the WER between a ground-truth
```
Comment (nikvaessen) [inline jiwer/measures.py:20]: Could you update the description of the file?

File: jiwer/measures.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+automatic speech recognition system
+"""
+
+import Levenshtein
+
+from typing import Union, List, Mapping
+
+import jiwer.transforms as tr
+
+__all__ = ["wer"]
```
Comment (nikvaessen) [inline jiwer/measures.py:31]: Could you update `_all_`?

File: jiwer/measures.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+    truth_str = ''.join(truth_chars)
+    hypothesis_str = ''.join(hypothesis_chars)
+
+    return truth_str, hypothesis_str
+
+def _get_operation_counts(
+    source_string: str,
+    destination_string: str
+) -> List[str]:
```
Comment (nikvaessen) [inline jiwer/measures.py:234]: Tuple[int, int, int, int]

**Comment by bartvanandel** [comment]
All done!
Thanks also for pointing me to `black`, I hadn't come across it before!

**Comment by bartvanandel** [comment]
BTW if you want I can squash this PR into 1 commit before you merge it, so the commit history stays tidy

**Comment by bartvanandel** [comment]
Sorry, messed up the pull request, I'll create another one once cleaned up

**Pull request closed without merging**